pkg/controller/controller_instance.go

```diff
@@ -203,10 +223,37 @@ func (c *controller) reconcileInstanceDelete(instance *v1alpha1.Instance) error
 	return nil
 }
 
+// isInstanceFailed returns whether the instance has a failed condition with
+// status true.
+func isInstanceFailed(instance *v1alpha1.Instance) bool {
```

Comment:
since this is failed ever at any point in the past. Does there need to be an eventual clean up code? Is the only way to resolve this to delete and recreate?